homeassistant/components/mikrotik/device_tracker.py

```diff
@@ -1,190 +1,136 @@
-                    interface = self.devices_arp[mac]["interface"]
-                    if not self.do_arp_ping(ip_address, interface):
-                        continue
-
-            attrs = {}
-            if mac in self.devices_dhcp and "host-name" in self.devices_dhcp[mac]:
-                hostname = self.devices_dhcp[mac].get("host-name")
-                if hostname:
-                    attrs["host_name"] = hostname
-
-            if self.devices_arp and mac in self.devices_arp:
-                attrs["ip_address"] = self.devices_arp[mac].get("address")
-
-            for attr in ATTR_DEVICE_TRACKER:
-                if attr in device and device[attr] is not None:
-                    attrs[slugify(attr)] = device[attr]
-            attrs["scanner_type"] = self.method
-            attrs["scanner_host"] = self.host
-            attrs["scanner_hostname"] = self.hostname
-            self.device_tracker[mac] = attrs
-
-    def do_arp_ping(self, ip_address, interface):
-        """Attempt to arp ping MAC address via interface."""
-        params = {
-            "arp-ping": "yes",
-            "interval": "100ms",
-            "count": 3,
-            "interface": interface,
-            "address": ip_address,
+        tracked[device] = MikrotikHubTracker(hub.api.devices[device], hub)
+
+        new_tracked.append(tracked[device])
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class MikrotikHubTracker(ScannerEntity):
+    """Representation of network device."""
+
+    def __init__(self, device, hub):
+        """Initialize the tracked device."""
+        self.device = device
+        self.hub = hub
+
+    @property
+    def is_connected(self):
+        """Return true if the client is connected to the network."""
+        if (
+            self.device.last_seen
+            and (dt_util.utcnow() - self.device.last_seen)
+            < self.hub.option_detection_time
+        ):
+            return True
+        return False
+
+    @property
+    def source_type(self):
+        """Return the source type of the client."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the client."""
+        return self.device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this device."""
+        return self.device.mac
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return self.hub.available
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        if self.is_connected:
+            return self.device.attrs
+
+    @property
+    def device_info(self):
+        """Return a client description for device registry."""
+        info = {
+            "connections": {(CONNECTION_NETWORK_MAC, self.unique_id)},
+            "identifiers": {(DOMAIN, self.unique_id)},
```

Comment:
It's more clear to use `device.mac` as unique id for the device. `unique_id` is the unique id of the entity. Those are not always the same although in this case they are.